src/main/kotlin/org/rust/lang/core/resolve/ref/RsResolveCache.kt

```diff
@@ -122,7 +122,7 @@ class RsResolveCache(messageBus: MessageBus) {
     private fun getCacheFor(element: PsiElement, dep: ResolveCacheDependency): ConcurrentMap<PsiElement, Any?> {
         return when (dep) {
             ResolveCacheDependency.LOCAL, ResolveCacheDependency.LOCAL_AND_RUST_STRUCTURE -> {
-                val owner = element.findModificationTrackerOwner()
+                val owner = element.findModificationTrackerOwner(strict = false)
```

Comment:
As I understand, you need this change to return `impl` item as modification tracker owner in case of `RsCachedImplItem`
And looks like we always get null here because `RsImplItem` doesn't implement `RsModificationTrackerOwner`.
And as a result we always return `rustStructureDependentCache`.
Maybe it's better to use `RUST_STRUCTURE` dependency for `RsCachedImplItem`?

Reply:
Wow, nice catch! You're right!